### PR TITLE
Add Local Service Discovery (BEP14)

### PR DESCRIPTION
**Pull request by hicom150**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Add Local Service Discovery (BEP14) support.

**Which issue (if any) does this pull request address?**
Fixes: https://github.com/webtorrent/torrent-discovery/issues/44

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -80,6 +84,12 @@ class Discovery extends EventEmitter {
       this.dht.on('peer', this._onDHTPeer)
       this._dhtAnnounce()
     }
+
+    if (opts.lsd === false) {
```
Comment (feross) [inline index.js:88]: Need to add `|| typeof LSD !== 'function'` for the browser case. The `browser: false` field you used in package.json tells Browserify to exclude the package and this is how to detect it.

File: package.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
     "url": "https://webtorrent.io"
   },
   "browser": {
-    "bittorrent-dht/client": false
+    "bittorrent-dht/client": false,
+    "bittorrent-lsd": false
```
Comment (feross) [inline package.json:12]: Good!

File: package.json [hunk 3]
```diff
@@ -17,6 +18,7 @@
   "dependencies": {
     "bittorrent-dht": "^10.0.0",
     "bittorrent-tracker": "^9.0.0",
+    "bittorrent-lsd": "^1.0.0",
```
Comment (feross) [inline package.json:21]: This package looks really good! Nice work!!

File: README.md [hunk 4]
```diff
@@ -11,14 +11,13 @@
...
-[bittorrent-tracker](https://www.npmjs.com/package/bittorrent-tracker) clients and exposes a
-single API for discovering BitTorrent peers via both discovery methods.
+This module bundles [bittorrent-dht](https://www.npmjs.com/package/bittorrent-dht),
+[bittorrent-tracker](https://www.npmjs.com/package/bittorrent-tracker) and [bittorrent-lsd](https://www.npmjs.com/package/bittorrent-lsd) clients and exposes a single API for discovering BitTorrent peers via both discovery methods.
 
 ## features
 
 - simple API
-- find peers from trackers and the DHT
+- find peers from trackers, DHT and LSD
```
Comment (feross) [inline README.md:20]: super nit-picky, but I like to put commas after "and" like this:

"find peers from trackers, DHT, and LSD"

**Review by feross — CHANGES_REQUESTED** [review]
AMAZING WORK! Just one important thing to fix for the browser case. Otherwise, LGTM!

File: package.json (hunk 3 above)
Comment (hicom150) [reply package.json:21]: Thank you! 😊

File: README.md (hunk 4 above)
Comment (hicom150) [reply README.md:20]: Changed! 👍

File: index.js (hunk 1 above)
Comment (hicom150) [reply index.js:88]: Fixed! 👍
I didn't know that, so learned for the next time 😉

**Review by feross — APPROVED** [review]

**Comment by feross** [comment]
@hicom150 One other thing – can you add `bittorrent-lsd` to the `webtorrent` npm organization? I think you can click "Add Package" on this page to do it: https://www.npmjs.com/settings/webtorrent/packages

**Comment by feross** [comment]
Published as 9.4.0

**Comment by feross** [comment]
Released in webtorrent 0.111.0

**Pull request merged**